src/panels/config/devices/ha-config-devices-dashboard.ts

```diff
@@ -235,37 +239,52 @@ export class HaConfigDeviceDashboard extends LitElement {
 
   private _columns = memoizeOne(
     (narrow: boolean, showDisabled: boolean): DataTableColumnContainer => {
-      const columns: DataTableColumnContainer = narrow
-        ? {
-            name: {
-              title: this.hass.localize(
-                "ui.panel.config.devices.data_table.device"
-              ),
-              main: true,
-              sortable: true,
-              filterable: true,
-              direction: "asc",
-              grows: true,
-              template: (name, device: DataTableRowData) => html`
-                ${name}
-                <div class="secondary">
-                  ${device.area} | ${device.integration}
-                </div>
-              `,
-            },
-          }
-        : {
-            name: {
-              title: this.hass.localize(
-                "ui.panel.config.devices.data_table.device"
-              ),
-              main: true,
-              sortable: true,
-              filterable: true,
-              grows: true,
-              direction: "asc",
-            },
-          };
+      const columns: DataTableColumnContainer = {
+        icon: {
+          title: "",
+          type: "icon",
+          template: (_icon, device) =>
+            device.domains.length
+              ? html`<img
+                  alt=""
```

Comment:
I maintain my view that this icon belongs in the integration column since that is what it is meant to convey.  It is not purely decorative here so hiding it from assistive technology is not appropriate.

As is, this creates 2 separate experiences:
1. Sighted users are reminded of the integration just by scanning the device name column
2. Screen reader users need to navigate 4 columns over to consume that information.

Just from a general UX standpoint, I don't see how it makes sense to put integration info inside a column labeled "Device" and then repeat it several columns later.

Reply:
> I maintain my view that this icon belongs in the integration column since that is what it is meant to convey.

From a pure data perspective, yes agree. But this ain't that. It is purely visual decoration.

Reply:
I'm more than happy to add a label of the integration to the icon, so screen reader users have that info in the first column too, but I don't think that helps you?

Moving it to the integration column is not an option in this case.

Reply:
> From a pure data perspective, yes agree. But this ain't that. It is purely visual decoration.

I get that's the intent, but no it isn't.  You could now take away the integration column and, at least for easily recognizable brands, no loss of information would occur.

> I'm more than happy to add a label of the integration to the icon, so screen reader users have that info in the first column too, but I don't think that helps you?

That would get closer to an equal experience, but you're right that it then becomes a confusing and chatty experience for a screen reader user, especially since that's the header column, which is why I don't think it belongs there to begin with.

> Moving it to the integration column is not an option in this case.

Okay now we're getting somewhere.  Why not?

Reply:
Because from a non-data perspective (but purely visual), it is odd to show an icon/visual representation in the middle of the screen.

>  You could now take away the integration column 

I don't think that is a fair comparison or argument to make. The column can't be removed.

Reply:
To put some context, in the future, we want to add the device picture. For now, we fallback to the integration image because we didn't have a device database yet. So it make sense to have it in the first column. 

We can add `alt=""` as it add no value for screen reader because the integration column is already here.
When we will be able to add the device image, it will add value so we can the appropriate `alt` for that.

Reply:
@frenck brand logos pretty much always add valuable information.  As much as you'd like to think of it as a pure decoration, you won't find any a11y analysts, graphic designers, marketing folks, etc. to take that same stance.  Companies spend big money to design them and plaster them everywhere for a reason.

So then the context needs to be considered when choosing the alternative text, and in this case a conundrum is created because it's questionably redundant and conveys information that doesn't match the table header.  Kept as is, I think `alt=""` is the best approach because of the cognitive confusion it would create to give it anything else.

And if it would be confusing to an alternative text consumer, then it will probably be confusing to others as well.  If I'm scanning a list of my devices and there's a graphic next to the name, I'd expect that it would be a picture of the device or at least the device brand.  And if I go to the entities list, am I also going to see the integration icon or the specific icon I chose for the entity?

@piitaya thanks, that's helpful context.  I think that just reinforces that the fallback should be the manufacturer logo though and not the integration, in which case the redundancy and position issues are lessened to a great degree.

Reply:
> And if I go to the entities list, am I also going to see the integration icon or the specific icon I chose for the entity?

We already show the entity icon in the entities data table, this was one of the few tables that didn't have an icon/image in the data table row

Reply:
> We already show the entity icon in the entities data table, this was one of the few tables that didn't have an icon/image in the data table row

I know - I was just making the point that it's inconsistent and confusing 😉

Reply:
> @frenck brand logos pretty much always add valuable information

Sure, I didn't say it didn't. However, in this context is a visual queue, a decoration, not data.

> And if it would be confusing to an alternative text consumer, then it will probably be confusing to others as well. If I'm scanning a list of my devices and there's a graphic next to the name

Sorry I have to disagree with most of your comments. Which is fine I guess 🤷

Reply:
> Sure, I didn't say it didn't. However, in this context is a visual queue, a decoration, not data.
 
To me you're contradicting yourself all over the place there.  Again, the author's intent is irrelevant.

> Sorry I have to disagree with most of your comments. Which is fine I guess 🤷

Yep, and that's sort of my point.  Not everyone consumes information like you or like me.  The whole point of accessibility is to be mindful of that wide range in the design.

Reply:
Sorry, stepping out of the convo. Your reasoning makes absolutely no sense to me whatsoever, nor do I understand any of the points you are trying to make. I think you are completely wrong and most of the things you mentioned above. I'm sure it's me for misunderstanding, however, I do not care enough about the image to continue the convo about it.